### ISSUE TITLE
Rainbow Screen on RPi Zero

### ISSUE THREAD

**Issue by nogleo** [boochow/micropython-raspberrypi#32]
I'm trying to make it work but no success... Follow the instructions and just rainbow... help please

**Comment by boochow** [comment]
Hi @nogleo, if you got the binary from [v1.0.0](https://github.com/boochow/micropython-raspberrypi/releases/tag/v1.0.0), the `main.py` need to be edited to enable the frame buffer console.
If you already edited the `main.py` as described in the `README.txt` and still couldn't get the frame buffer enabled, try to replace `main.py` with this [single-file alternative](https://gist.github.com/boochow/939c2fc9d3fb76cc10cf97a49f08f5e4).

**Comment by nogleo** [comment]
![Screenshot from 2020-08-05 10-57-07](https://user-images.githubusercontent.com/62610230/89422077-3b6f8d80-d70b-11ea-9fa4-23eaab10a077.png)
sadly I've tried both cases and got nothing... 

maybe I'm just noobing on something

**Comment by nogleo** [comment]
is it really all I have to put into de SD to make it work?

**Comment by boochow** [comment]
Your files seem OK. 
To test whether the boot process is succeeded or not, try this:

The attached file is a simple bare-metal program that blinks Pi's LED eternally. (source code is [here](https://github.com/boochow/bare_matal_rpi_zero/tree/master/led_blynk).)
Extract `blink_led.img` from zip and copy it to your SD card.
Edit `config.txt` and replace the line `kernel=firmware.img` to `kernel=blink_led.img` then save the file.
If your Pi's LED blinks, then your `bootcode.bin` and `start.elf` have no problem.

I know there are some compatibility issues between some SD cards and the SD card driver in micropython. Using other SD cards may solve the problem.

[blink_led.zip](https://github.com/boochow/micropython-raspberrypi/files/5029545/blink_led.zip)

**Comment by nogleo** [comment]
Tested it with 2 different SDcards ando no hope, not even the rainbow was appearing anymore

**Comment by boochow** [comment]
I made the disk image of my SD card that runs correctly. 
Try writing this image to your SD card with [Raspberry Pi Imager](https://www.raspberrypi.org/downloads/).
The disk size will become 128MB regardless of the actual size of your SD card, but don't mind. It is because I set the size of the first partition to 128MB.
 
[upyrpi_usb.zip](https://github.com/boochow/micropython-raspberrypi/files/5034794/upyrpi_usb.zip)

**Comment by nogleo** [comment]
IT'S ALIIIIIIVE!!!

Thank you very much!!! I'm still not sure what is the thing that made it work hahahahaha.
Maybe working on linux has something to do with that? 
Actually the usb host is not working and gives me erros

Just one question now, can it work through repl?

**Comment by boochow** [comment]
Congratulations!
Some USB keyboards works, others not. If you could find one that works, it would work with REPL.
I have tested several USB keyboards, but have not yet found the conditions that determine whether a keyboard works or not.
I'm using this keyboard for testing.  =>  https://www.amazon.com/dp/B00W7WSIKS/

**Comment by nogleo** [comment]
when I ask about repl I mean run the repl in the PC like with a ESP32. Is it possible?

**Comment by boochow** [comment]
Yeah, it is possible, but you need a USB-UART adapter. Most of the ESP32 development boards have a USB-serial converter chip on board, but RPi doesn't. An external USB-serial module should be connected to your PC and Pi like the attached image.
![rpi-zero-serial](https://user-images.githubusercontent.com/8764949/89591663-bbb9fe00-d885-11ea-8431-729929f7ada8.jpg)
